### PR TITLE
[rebase] Add support for buildpack test command

### PR DESCRIPTION
**Pull request by michaelshobbs**
closes #271

### CODE REVIEW

File: include/buildpack.bash [hunk 1]
```diff
@@ -21,6 +21,54 @@ _envfile-parse() {
...
+		# shellcheck disable=SC2154
+		selected_path="$buildpack_path/custom"
+		rm -rf "$selected_path"
+
+		IFS='#' read -r url commit <<< "$BUILDPACK_URL"
+		buildpack-install "$url" "$commit" custom &> /dev/null
+		# unprivileged_user & unprivileged_group defined in outer scope
+		# shellcheck disable=SC2154
+		chown -R "$unprivileged_user:$unprivileged_group" "$buildpack_path/custom"
+		selected_name="$(unprivileged "$selected_path/bin/detect" "$build_path" || true)"
```
Comment (michaelshobbs) [inline include/buildpack.bash]: this sets off my variable scope alarm but the alternative is to return an array i guess?

**Comment by michaelshobbs** [comment]
@dzaporozhets mind taking a look?

**Comment by dzaporozhets** [comment]
@michaelshobbs looks good to me

**Comment by blackjid** [comment]
Can't wait!! Thanks @dzaporozhets and @michaelshobbs!

**Comment by blackjid** [comment]
@dzaporozhets have you thought how multiple buildpacks should be handled? 

The buildpack-multi should have to get support for executing the `test-compile` and `test` scripts from all the buildpacks??

It's kind of confusing because herokuish rely on buildpack-multi to execute multiple buildpacks, but heroku now support that natively (probably this is a discussion for another issue)

Right now is not posible to use `herokuish buildpack test` having a `.buildpack` file becasue it complains that there is no `test-compile` file

thanks!

**Comment by markpundsack** [comment]
@blackjid Good point. Sounds like something that needs to be added to the multi-buildpack itself.

File: tests/unit/tests.sh [hunk 2]
```diff
@@ -71,7 +71,7 @@ T_procfile-types() {
     expected="Procfile declares types -> web, worker"
     actual="$(procfile-types invalid-proc | tail -1)"
 
-    if [[ $actual != $expected ]]; then
+    if [[ $actual != "$expected" ]]; then
```
Comment (josegonzalez) [inline tests/unit/tests.sh]: Do we need quotes around `$actual` as well?

File: include/buildpack.bash [hunk 3]
```diff
@@ -178,11 +196,28 @@ buildpack-execute() {
...
+buildpack-test() {
+	declare desc="Build and run tests for an application using installed buildpacks"
+	ensure-paths
+	[[ "$USER" ]] || randomize-unprivileged
+	buildpack-setup > /dev/null
+	_select-buildpack
+
+	if [[ ! -f "$selected_path/bin/test-compile" ]] || [[ ! -f "$selected_path/bin/test" ]]; then
+		echo "Selected buildpack does not support test feature"
+		return
```
Comment (josegonzalez) [inline include/buildpack.bash]: Should this exit 1?

File: tests/unit/tests.sh (hunk 2 above)
Comment (michaelshobbs) [reply tests/unit/tests.sh]: Not according to `shellcheck`

https://github.com/koalaman/shellcheck/wiki/SC2053

File: include/buildpack.bash (hunk 3 above)
Comment (michaelshobbs) [reply include/buildpack.bash]: Probably `exit 1`. Fixed

File: tests/unit/tests.sh (hunk 2 above)
Comment (josegonzalez) [reply tests/unit/tests.sh]: I would just add them there, seems silly to not.

Comment (michaelshobbs) [reply tests/unit/tests.sh]: done

**Comment by blackjid** [comment]
@markpundsack I've been playing with this. I forked the original multi buildpack repo and I added initial support for tests. https://github.com/platanus/heroku-buildpack-multi/tree/feat/testpack_support

**Comment by markpundsack** [comment]
@blackjid Awesome!

**Comment by dzaporozhets** [comment]
@michaelshobbs thank you

**Pull request merged**